kernel/thread.c

```diff
@@ -377,10 +377,72 @@ long myst_syscall_wait4(
         ((options & WNOHANG) == WNOHANG) ? "NO HANG" : "HANG");
 #endif
 
-    if (rusage)
+    ECHECK(myst_wait(pid, wstatus, NULL, options, rusage));
+
+done:
+    return ret;
+}
+
+long myst_syscall_waitid(
+    idtype_t idtype,
+    id_t id,
+    siginfo_t* infop,
+    int options)
+{
+    long ret = 0;
+
+    if (options & ~(WEXITED | WSTOPPED | WCONTINUED | WNOHANG))
```

Comment:
WNOWAIT is missing here. And also some Linux specific additional flags - __WNOTHREAD, __WALL and __WCLONE.

Reply:
I need to confirm with Mike again, but we will probably end up only supporting WEXITED and WNOHANG.